### PR TITLE
Fix further issues with TLS/SSL

### PR DESCRIPTION
**Pull request by g-andrade**
Following your recent changes for OTP 23 support in 66415cf63a, I found some other problems which this PRs proposes fixing.

- broken compilation due to failing `ct_expand` parse transform, fixed in 92c0f4c
- crashing HTTPS requests in OTP 19, fixed in 8eddf7c
- occasional TLS negotiation errors in OTP 20, fixed in 533889d

Cheers,

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -20,6 +25,7 @@
         {mimerl, "~>1.1"},
         {certifi, "2.5.2"},
         {metrics, "1.0.1"},
+        {parse_trans, "3.3.0"},
```
Comment (benoitc) [inline rebar.config:28]: Is this needed? parse_trans is coming with certifi normally

File: src/hackney_ssl.erl [hunk 2]
```diff
@@ -88,16 +109,14 @@ decoded_cacerts() ->
   ct_expand:term(
     lists:foldl(fun(Cert, Acc) ->
                     Dec = public_key:pkix_decode_cert(Cert, otp),
-                    [extract_public_key_info(Dec) | Acc]
+                    [hackney_ssl_certificate:public_key_info(Dec) | Acc]
                 end, [], certifi:cacerts())
    ).
 
 
-extract_public_key_info(Cert) ->
```
Comment (benoitc) [inline src/hackney_ssl.erl:96]: It's fine but curious why moving it to another module?

File: rebar.config (hunk 1 above)
Comment (benoitc) [reply rebar.config:28]: it's fine i will commit with it

**Comment by benoitc** [comment]
thanks for the patch!

File: src/hackney_ssl.erl (hunk 2 above)
Comment (benoitc) [reply src/hackney_ssl.erl:96]: sorry found the explanation in the commit.... :)

File: rebar.config (hunk 1 above)
Comment (g-andrade) [reply rebar.config:28]: I just meant to be explicit, since `hackney` started depending directly on it.

**Pull request merged**